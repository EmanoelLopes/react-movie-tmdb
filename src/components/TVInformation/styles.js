import styled from 'styled-components';
import { colors } from 'utils/styled';

export const TVInformationContainer = styled.section`
  p { margin-bottom: 15px; }
  a { color: ${colors.mainColor}; }
`;
