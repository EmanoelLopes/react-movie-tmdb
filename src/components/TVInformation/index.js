import React from 'react';
import { string, number } from 'prop-types';
import { TVInformationContainer } from './styles';

const TVInformation = props => {
  const {
    firstAirDate,
    numberOfEpisodes,
    seasonNumber,
    episodeNumber,
    episodeName,
    homepage,
  } = props;

  return (
    <TVInformationContainer>
      <p><strong>First air date: </strong> {firstAirDate}</p>
      <p><strong>Number of episodes: {numberOfEpisodes}</strong></p>
      <p><strong>Current season: {seasonNumber}</strong></p>
      <p>
        <strong>Last episode to air: &nbsp;
          {episodeNumber} - {episodeName} 
        </strong>
      </p>
      <p>
        <strong>Official Site: </strong>
        <a
          href={`${homepage}`}
          target="_blank"
          rel="noopener noreferrer">{homepage}
        </a>
      </p>
    </TVInformationContainer>
  );
};

TVInformation.propTypes = {
  firstAirDate: string,
  numberOfEpisodes: number,
  seasonNumber: number,
  episodeNumber: number,
  episodeName: string,
  homepage: string,
};

TVInformation.defaulProps = {
  firstAirDate: '',
  numberOfEpisodes: 0,
  seasonNumber: 0,
  episodeNumber: 0,
  episodeName: 0,
  homepage: '',
};

export default TVInformation;
