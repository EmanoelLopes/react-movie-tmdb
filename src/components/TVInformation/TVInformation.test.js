import React from 'react';
import { shallow } from 'enzyme';
import TVInformation from 'components/TVInformation';
import locale from 'lang';

locale.init();

let wrapped;
let defaultProps = {
  firstAirDate: '',
  numberOfEpisodes: 0,
  numberOfSeasons: 0,
  seasonNumber: 0,
  episodeNumber: 0,
  episodeName: '',
  nextEpisodeNumber: 0,
  nextEpisodeName: '', 
  homepage: '',
};

beforeEach(() => {
  wrapped = shallow(<TVInformation {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<TVInformation />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
