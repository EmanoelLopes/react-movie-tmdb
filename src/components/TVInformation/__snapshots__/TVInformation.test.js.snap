// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TVInformation /> should match snapshot 1`] = `
<styled.section>
  <p>
    <strong>
      First air date: 
    </strong>
     
  </p>
  <p>
    <strong>
      Number of episodes: 
      0
    </strong>
  </p>
  <p>
    <strong>
      Current season: 
      0
    </strong>
  </p>
  <p>
    <strong>
      Last episode to air: Â 
      0
       - 
    </strong>
  </p>
  <p>
    <strong>
      Official Site: 
    </strong>
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    />
  </p>
</styled.section>
`;
